protocols/kad/src/behaviour/test.rs

```diff
@@ -1606,3 +1606,51 @@ fn get_providers_limit_n_2() {
 fn get_providers_limit_n_5() {
     get_providers_limit::<5>();
 }
+
+// Test that nodes respond with K amount of peers even when replication factor is set lower than K.
+#[test]
+fn get_closest_peers_should_return_up_to_k_peers() {
```

Comment:
Hi @guillaumemichel , is this test what you had in mind?

Reply:
yes!